android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java

```diff
@@ -440,6 +450,36 @@ public static boolean executeHardwareKeystroke(
     return false;
   }
 
+  /**
+   * Handle the globe key longpress action. Currently just for INAPP only
+   * @param context
+   * @param keyboard
+   */
+  private static void doGlobeKeyLongpressAction(Context context, KeyboardType keyboard) {
+    if (keyboard == KeyboardType.KEYBOARD_TYPE_INAPP && InAppKeyboard != null) {
+      if (InAppKeyboard.keyboardPickerEnabled) {
+        showKeyboardPicker(context, keyboard);
+      }
+    }
+  }
+
+  /**
+   * Handle the globe key shortpress action. Currently just for INAPP only
+   * @param context
+   * @param keyboard
+   */
+  private static void doGlobeKeyShortpressAction(Context context, KeyboardType keyboard) {
+    if (keyboard == KeyboardType.KEYBOARD_TYPE_INAPP && InAppKeyboard != null) {
+      if (InAppKeyboard.keyboardPickerEnabled && inappKbGlobeKeyAction == GlobeKeyAction.GLOBE_KEY_ACTION_SHOW_MENU) {
+        showKeyboardPicker(context, KeyboardType.KEYBOARD_TYPE_INAPP);
+      } else if (inappKbGlobeKeyAction == GlobeKeyAction.GLOBE_KEY_ACTION_SWITCH_TO_NEXT_KEYBOARD) {
+        switchToNextKeyboard(context);
+      } else {
+        switchToNextKeyboard(context);
+      }
```

Comment:
```suggestion
      } else { /* inappKbGlobeKeyAction == GlobeKeyAction.GLOBE_KEY_ACTION_SWITCH_TO_NEXT_KEYBOARD */
        switchToNextKeyboard(context);
      }
```

Is that really right?

Reply:
yeah, that should be fine too.